blocks/models/huggingface/base.py

```diff
@@ -24,55 +36,92 @@ def __init__(self, id: str, config: HuggingfaceConfig):
+            self.model = model
+        else:
+            print("     ├ ℹ️ No local model found")
+
+        return execution_order + 1
 
     def load_remote(self):
-        repo_name = self.config.user_name + "/" + self.config.repo_name
-        try:
-            self.model = load_pipeline(repo_name)
-        except:
-            print("❌ No model found in huggingface repository")
+        if self.model is None:
+            model = safe_load_pipeline(f"{self.config.user_name}/{self.id}")
+            if model:
+                self.model = model
+            else:
+                print(
+                    f"     ├ ℹ️ No fitted model found remotely, loading pretrained foundational model: {self.config.pretrained_model}"
+                )
+                self.model = safe_load_pipeline(self.config.pretrained_model)
+                self.pretrained = True
 
     def fit(self, dataset: List[str], labels: Optional[pd.Series]) -> None:
-
         train_dataset, val_dataset = train_test_split(
             pd.DataFrame({Const.input_col: dataset, Const.label_col: labels}),
             test_size=self.config.val_size,
         )
 
-        model = run_training_pipeline(
+        trainer = run_training_pipeline(
             from_pandas(train_dataset, self.config.num_classes),
             from_pandas(val_dataset, self.config.num_classes),
             self.config,
+            self.pipeline_id,
+            self.id,
         )
-        self.model = load_pipeline(model)
+        self.model = safe_load_pipeline(trainer.model)
+        #     AutoModelForSequenceClassification.from_pretrained(trainer.model)
+        # )
+        self.trainer = trainer
         self.trained = True
 
     def predict(self, dataset: pd.DataFrame) -> pd.DataFrame:
-        if self.model:
-            model = self.model
-        else:
-            print(
-                f"❌ No fitted model, using inference on pretrained foundational model :{self.config.pretrained_model}"
-            )
-            model = load_pipeline(self.config.pretrained_model)
-
         return run_inference_pipeline(
-            model,
-            from_pandas(dataset, self.config.num_classes),
-            self.huggingface_config,
+            self.model,
+            from_pandas(
+                pd.DataFrame({Const.input_col: dataset}), self.config.num_classes
+            ),
+            self.config,
         )
 
     def is_fitted(self) -> bool:
-        return self.model is not None
-
-
-def from_pandas(df: pd.DataFrame, num_classes: int) -> Dataset:
-    return Dataset.from_pandas(
-        df,
-        features=Features(
-            {
-                Const.input_col: Value("string"),
-                Const.label_col: ClassLabel(num_classes),
-            }
-        ),
-    )
+        return self.pretrained is False
+
+    def save(self) -> None:
+        pass
+        # if self.config.save and self.trained:
+        #     path = f"{Const.output_path}/{self.pipeline_id}/{self.id}"
```

Comment:
is this needed?:)

Reply:
We need to keep save(self) in huggingface otherwise it will do the default and save a pytorch object. I took out the commented part!

Reply:
ahh I see yes!